sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -2542,9 +2823,34 @@ def run(self):
         for thread in self.threads:
             self.log_notice("Started thread {}".format(thread.getName()))
 
-        self.stop_event.wait()
+        sel, asic_context  = self.subscribe_appl_port_table()
+        self.log_notice("Subscribed to appl {} update".format(swsscommon.APP_PORT_TABLE_NAME))
 
-        self.log_info("Stop daemon main loop")
+        generate_sigabrt = False
+
+        while not self.stop_event.is_set() and not generate_sigabrt:
+            (state, _) = sel.select(port_mapping.SELECT_TIMEOUT_MSECS)
+            if state == swsscommon.Select.TIMEOUT:
```

Comment:
seems to me following condition `if state != swsscommon.Select.OBJECT:` is enough and this one is not needed?

Reply:
I have kept this check to avoid flooding of logs when select times out